### PR TITLE
Changes to apps needed by nuttx PR 8885

### PR DESCRIPTION
**Pull request by patacongo**
## Summary

ostest contains some logic that depends on internal implementation of signal sets and ostest must be updated to match those changes.

This PR will not pass CI because it depends on #apache/nuttx/8885

## Impact

There is no particular impact from this PR.  This PR is the result of impact from nuttx 8885.

## Testing

Tested with https://github.com/apache/nuttx/pull/8885

### CODE REVIEW

**Review by pkarashchenko — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

**Comment by pkarashchenko** [comment]
@patacongo could you please mark this as "ready for review"?

File: testing/ostest/ostest.h [hunk 1]
```diff
@@ -190,6 +185,10 @@ void timedwait_test(void);
 
 void sigprocmask_test(void);
 
+/* sighelper.c **************************************************************/
+
+bool sigset_isequal(FAR const sigset_t *right, FAR const sigset_t *left);
```
Comment (xiaoxiang781216) [inline testing/ostest/ostest.h:190]:
```suggestion
bool sigset_isequal(FAR const sigset_t *left, FAR const sigset_t *right);
```

File: testing/ostest/sighelper.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+#include <stdbool.h>
+#include <signal.h>
+
+#include "ostest.h"
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+bool sigset_isequal(FAR const sigset_t *right, FAR const sigset_t *left)
```
Comment (xiaoxiang781216) [inline testing/ostest/sighelper.c:36]:
```suggestion
bool sigset_isequal(FAR const sigset_t *left, FAR const sigset_t *right)
```

File: testing/ostest/ostest.h (hunk 1 above)
Comment (patacongo) [reply testing/ostest/ostest.h:190]: Done

File: testing/ostest/sighelper.c (hunk 2 above)
Comment (patacongo) [reply testing/ostest/sighelper.c:36]: Done

**Review by pkarashchenko — APPROVED** [review]

File: testing/ostest/posixtimer.c [hunk 3]
```diff
@@ -170,8 +173,9 @@ void timer_test(void)
     }
 
 #ifndef SDCC
-  printf("timer_test: oact.sigaction=%p oact.sa_flags=%x oact.sa_mask=%jx\n",
-          oact.sa_sigaction, oact.sa_flags, (uintmax_t)oact.sa_mask);
+  printf("timer_test: oact.sigaction=%p oact.sa_flags=%x "
+         "oact.sa_mask=" SIGSET_FMT "\n",
+          oact.sa_sigaction, oact.sa_flags, SIGSET_ELEM(&oact.sa_mask));
```
Comment (pkarashchenko) [inline testing/ostest/posixtimer.c:178]:
```suggestion
  printf("timer_test: oact.sigaction=%p oact.sa_flags=%x "
         "oact.sa_mask=" SIGSET_FMT "\n",
         oact.sa_sigaction, oact.sa_flags, SIGSET_ELEM(&oact.sa_mask));
```
BTW, very strange why it is hidden under `#ifndef SDCC`

**Pull request merged**